src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -150,6 +152,34 @@ impl ManagedBarrierState {
         self.create_mview_progress.clear();
     }
 
+    /// Notify unexpected actor exit with given `actor_id`.
+    pub(crate) fn notify_exit(&mut self, actor_id: ActorId) {
+        for (_, barrier_state) in &mut self.epoch_barrier_state_map {
+            match barrier_state.inner {
+                ManagedBarrierStateInner::Issued {
+                    ref remaining_actors,
+                    ref mut collect_notifier,
+                } => {
```

Comment:
I think we should also handle the `Stashed` case. Unfortunately, we don't have the `collect_notifier` here.

Considering this, how about adding a `Failed` state? If `transfrom_to_issued` meets `Failed`, it should directly send the `Err` to the channel. If multiple actors fail, the latter ones will also find this state has failed and do nothing.

Reply:
> 
> If multiple actors fail, the latter ones will also find this state has failed and do nothing.

So the `Failed` state should be a status of `ManagedBarrierState` rather than `BarrierState` and don't bind to any epoch. I will give it a try.

Reply:
I added a field `exited_actors` rather than a simple `Failed` state for compatibility, if we want to support partial checkpoint in the future.